tests/GamepadService-spec.js

```diff
@@ -112,3 +114,84 @@ describe("Consumer", () => {
     expect(consumer.hasGesture).toBe(false);
   });
 });
+
+describe("GamingDevice", () => {
```

Comment:
This should be probably be in its own file "GamindDevice-spec.js". Similarly for tests specific for GamingDeviceButton.js.